### PR TITLE
Add feature to copy chat messages to clipboard

### PR DESCRIPTION
**Pull request by jjbillings**
my Habitica User-ID: d5de2b4c-caa6-4927-97b2-4ad0bda7e85a (Jack Billings)

### CODE REVIEW

**Comment by jjbillings** [comment]
Hello! I was a little confused about the output of the test. It appears as thought the tests failed due to some stuff with roboelectric. I re-cloned the repository and ran the test without making any changes to the code and the test failed with the same results. Could this be a problem with how I have it configured on my computer? Or possibly an error in the current source code available in the repo?

Any advice would be appreciated.

File: Habitica/src/main/java/com/habitrpg/android/habitica/ui/fragments/social/ChatListFragment.java [hunk 1]
```diff
@@ -160,6 +165,16 @@ private void markMessagesAsSeen() {
     }
 
     @Subscribe
+    public void onEvent(CopyChatMessageCommand cmd)
+    {
+        ClipboardManager clipMan = (ClipboardManager)getActivity().getSystemService(Context.CLIPBOARD_SERVICE);
+        ClipData chatMessage = ClipData.newPlainText("Chat Message",cmd.chatMessage.text);
+        clipMan.setPrimaryClip(chatMessage);
+        Toast.makeText(getActivity().getApplicationContext(),
```
Comment (vIiRuS) [inline Habitica/src/main/java/com/habitrpg/android/habitica/ui/fragments/social/ChatListFragment.java]: This should be a snackbar instead.

File: Habitica/res/menu/chat_message.xml [hunk 2]
```diff
@@ -10,6 +10,11 @@
         android:title="@string/send_pm" />
     <group>
         <item
+                android:id="@+id/menu.chat.copy"
+                android:icon="@drawable/ic_action_warning"
+                app:showAsAction="always|withText"
+                android:title="@string/copy_chat_message" />
```
Comment (vIiRuS) [inline Habitica/res/menu/chat_message.xml]: This string isn't found in the strings.xml file. did you maybe forget to commit it?

Comment (jjbillings) [inline Habitica/res/menu/chat_message.xml]: Sorry! When I merged the latest changes from the repo that was a conflict and it got removed. committing it now

**Comment by vIiRuS** [comment]
I merged this in manually and github didn't recognize that, so I'll have to close this manually now. Thanks for your work!

I noted this as a contribution in your habitica profile

**Pull request closed without merging**